### PR TITLE
#166272339 Feature Handle User Profile

### PR DESCRIPTION
**Pull request by despeauxz**
#### What does this PR do?
This PR adds two endpoint
- GET `localhost:3000/api/v1/profiles/{username}`
- PUT `localhost:3000/api/v1/users`
#### How should this be manually tested?
Navigate to the endpoints on Postman
- GET `localhost:3000/api/v1/profiles/{username}`
- PUT `localhost:3000/api/v1/user`, specify the Bearer token to specify Profile to update
#### Any background context you want to provide?
- Documented endpoints on Swagger
- Write unit test for endpoints
#### What are the relevant pivotal tracker stories?
[#166272339](https://www.pivotaltracker.com/story/show/166272339)
#### Screenshots (if appropriate)
N/A
#### Questions:
N/A

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -15,7 +15,7 @@
     "clean": "rm -rf dist",
     "coverage": "nyc report --reporter=text-lcov | coveralls",
     "pretest": "rm -rf ./server/database_test.sqlite3 && NODE_ENV=test npm run refreshdb",
-    "test": "NODE_ENV=test nyc --reporter=html --reporter=text --require @babel/register --env=test mocha --timeout 10000 ./server/tests/*.js --exit"
+    "test": "NODE_ENV=test nyc --reporter=html --reporter=text --require @babel/register --env=test mocha --timeout 10000 ./server/tests/ --exit"
```
Comment (fidelisojeah) [inline package.json]: perhaps 
```suggestion
    "test": "NODE_ENV=test nyc --reporter=html --reporter=text --require @babel/register --env=test mocha --timeout 10000 ./server/tests/*.test.js --exit"
```
?

File: server/controllers/profile.js [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+  static async updateProfile(req, res, next) {
+    try {
+      const { id } = req.decoded;
+      const profileDetails = await validateProfileDetails(req.body);
+
+      const {
+        firstname, lastname, bio, avatar, phone, location
+      } = profileDetails;
+
+      const profile = await Profile.findOne({
```
Comment (fidelisojeah) [inline server/controllers/profile.js]: Try something like:
```suggestion
      const profile = await req.user.getProfile();
```

I've not tested this but I believe may have what you need

File: server/routes/api/user.js [hunk 3]
```diff
@@ -0,0 +1,11 @@
+import express from 'express';
+import trim from '@middlewares/trim';
+import ProfileController from '@controllers/profile';
+import Token from '@helpers/Token';
+
+const userRouter = express.Router();
+
+userRouter.put('/', Token.authorize, trim, ProfileController.updateProfile);
```
Comment (fidelisojeah) [inline server/routes/api/user.js]: in this situation, ignore the readme.

user `PUT /users/` to update the user instead (so move this logic to the `userRouter`
simlar for `GET /users/` if you are the one working on it

File: server/tests/factory/userFactory.js [hunk 4]
```diff
@@ -0,0 +1,24 @@
+import faker from 'faker';
```
Comment (fidelisojeah) [inline server/tests/factory/userFactory.js]: I believe there's already a `user-factory.js` file that does this

Comment (fidelisojeah) [reply server/tests/factory/userFactory.js]: also, you could modify the `createTestUser` factory to make a call to  `createProfileDetails` then have another function that would be something like `createTestUserWithoutProfile`.

**Review by fidelisojeah — CHANGES_REQUESTED** [review]
Great job.

Just a few changes here and there and you would be fine

**Review by EMacco — DISMISSED** [review]

**Review by EMacco — APPROVED** [review]

**Review by GerrardE — APPROVED** [review]
Good job on implementing feedback on swagger api documentation as discussed some minutes ago.

**Review by fidelisojeah — APPROVED** [review]

**Pull request merged**